app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/NoteAPI.java

```diff
@@ -119,4 +120,140 @@ public ArrayList<Long> findNotes(String query) {
+                NoteInfoField> fields) {
+            this.noteId = noteId;
+            this.modelName = modelName;
+            this.tags = tags;
+            this.fields = fields;
+        }
+
+        public long getNoteId() {
+            return noteId;
+        }
+
+        public String getModelName() {
+            return modelName;
+        }
+
+        public List<String> getTags() {
+            return tags;
+        }
+
+        public Map<String, NoteInfoField> getFields() {
+            return fields;
+        }
+    }
+
+    static class Model {
+        private final long modelId;
+        private final String modelName;
+        private final String[] fieldNames;
+
+        public Model(long modelId, String modelName, String[] fieldNames) {
+            this.modelId = modelId;
+            this.modelName = modelName;
+            this.fieldNames = fieldNames;
+        }
+
+        public long getModelId() {
+            return modelId;
+        }
+
+        public String getModelName() {
+            return modelName;
+        }
+
+        public String[] getFieldNames() {
+            return fieldNames;
+        }
+    }
+
+    public List<NoteInfo> notesInfo(ArrayList<Long> noteIds) throws Exception {
+        List<NoteInfo> notesInfoList = new ArrayList<>();
+        String nidQuery = "nid:" + TextUtils.join(",", noteIds);
+        Map<Long, Model> cache = new HashMap<>();
+
+        Cursor cursor = this.resolver.query(
+                FlashCardsContract.Note.CONTENT_URI,
+                NOTES_INFO_PROJECTION,
+                nidQuery,
+                null,
+                null,
+                null
+                );
+
+        if (cursor == null) {
+            return null;
+        }
+
+        try (cursor) {
+            while (!cursor.moveToNext()) {
+
+                int idIdx = cursor.getColumnIndexOrThrow(FlashCardsContract.Note._ID);
+                int midIdx = cursor.getColumnIndexOrThrow(FlashCardsContract.Note.MID);
+                int tagsIdx = cursor.getColumnIndexOrThrow(FlashCardsContract.Note.TAGS);
+                int fldsIdx = cursor.getColumnIndexOrThrow(FlashCardsContract.Note.FLDS);
+
+                long id = cursor.getLong(idIdx);
+                long mid = cursor.getLong(midIdx);
+                List<String> tags = Arrays.asList(Utility.splitTags(cursor.getString(tagsIdx)));
+                String[] fieldValues = Utility.splitFields(cursor.getString(fldsIdx));
+                Model model = null;
+
+                if (cache.containsKey(mid)) {
+                    model = cache.get(mid);
+                }
+                else {
+                    String[] fieldNames = api.getFieldList(mid);
+                    String modelName = api.getModelName(mid);
+
+                    if (fieldNames.length != fieldValues.length) {
+                        // shouldn't happen
+                        throw new Exception("fieldNames.length != fieldValues.length");
```

Comment:
This is thrown within a try without a catch, does it actually do anything?
Does this ever happen?

Reply:
Unless there's a problem internally with the anki database I don't think this should happen, will remove